File: FloatingShortcutsWear/src/main/java/net/geekstools/floatshort/PRO/BindServices.java
package net.geekstools.floatshort.PRO;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.app.Service;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.IBinder;
import android.text.Html;
import android.widget.RemoteViews;

import net.geekstools.floatshort.PRO.Util.Functions.FunctionsClass;
import net.geekstools.floatshort.PRO.Util.Functions.PublicVariable;

public class BindServices extends Service {

    RemoteViews remoteNotfication;

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        PublicVariable.contextStatic = getApplicationContext();
        FunctionsClass functionsClass = new FunctionsClass(getApplicationContext());
        if (functionsClass.returnAPI() < 26) {
            startForeground(333, bindServiceLow());
        } else {
            startForeground(333, bindServiceHigh());
        }

        return START_STICKY;
    }

    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    protected Notification bindServiceLow() {
        Notification.Builder mBuilder = new Notification.Builder(this);

        Bitmap bM = BitmapFactory.decodeResource(getResources(), R.drawable.ic_launcher);

        mBuilder.setContent(remoteNotfication);
        mBuilder.setContentTitle(Html.fromHtml("<b><font color='" + getColor(R.color.light) + "'>" + getResources().getString(R.string.bindTitle) + "</font></b>"));
        mBuilder.setContentText(Html.fromHtml("<font color='" + getColor(R.color.light) + "'>" + getResources().getString(R.string.bindDesc) + "</font>"));
        mBuilder.setTicker(getResources().getString(R.string.app_name));
        mBuilder.setSmallIcon(R.drawable.ic_notification);
        mBuilder.setLargeIcon(bM);
        mBuilder.setAutoCancel(false);
        mBuilder.setColor(getColor(R.color.default_color));
        mBuilder.setPriority(Notification.PRIORITY_MAX);

        Intent Main = new Intent(this, Configurations.class);
        PendingIntent mainPV = PendingIntent.getActivity(this, 0, Main, PendingIntent.FLAG_UPDATE_CURRENT);

        mBuilder.setContentIntent(mainPV);

        return mBuilder.build();
    }

    protected Notification bindServiceHigh() {
        NotificationManager notificationManager = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);

        NotificationChannel notificationChannel = new NotificationChannel(getPackageName(), getString(R.string.app_name), NotificationManager.IMPORTANCE_MAX);
        notificationManager.createNotificationChannel(notificationChannel);

        Notification.Builder mBuilder = new Notification.Builder(this);
        mBuilder.setContent(remoteNotfication);
        mBuilder.setContentTitle(Html.fromHtml("<b><font color='" + getColor(R.color.light) + "'>" + getResources().getString(R.string.bindTitle) + "</font></b>"));
        mBuilder.setContentText(Html.fromHtml("<font color='" + getColor(R.color.light) + "'>" + getResources().getString(R.string.bindDesc) + "</font>"));
        mBuilder.setTicker(getResources().getString(R.string.app_name));
        mBuilder.setSmallIcon(R.drawable.ic_notification);
        mBuilder.setLargeIcon(BitmapFactory.decodeResource(getResources(), R.drawable.ic_launcher));
        mBuilder.setAutoCancel(false);
        mBuilder.setColor(getColor(R.color.default_color));
        mBuilder.setChannelId(getPackageName());
        mBuilder.setContentIntent(PendingIntent.getActivity(this, 0,
                new Intent(this, Configurations.class),
                PendingIntent.FLAG_UPDATE_CURRENT));

        //notificationManager.notify(666, mBuilder.build());
        return mBuilder.build();
    }
}

File: FloatItWatch/src/main/java/net/geekstools/floatshort/PRO/BindServices.kt
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 4/27/20 10:24 AM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */
package net.geekstools.floatshort.PRO

import android.app.*
import android.content.Context
import android.content.Intent
import android.graphics.BitmapFactory
import android.os.Build
import android.os.IBinder
import android.text.Html
import androidx.annotation.RequiresApi

class BindServices : Service() {

    override fun onBind(intent: Intent?): IBinder? {

        return null
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            startForeground(333, bindServiceHigh())
        } else {
            startForeground(333, bindServiceLow())
        }

        return Service.START_STICKY
    }

    private fun bindServiceLow(): Notification {
        val notificationBuilder = Notification.Builder(this)

        val bitmap = BitmapFactory.decodeResource(resources, R.drawable.ic_launcher)

        notificationBuilder.setContentTitle(Html.fromHtml("<b><font color='" + getColor(R.color.light) + "'>" + resources.getString(R.string.bindTitle) + "</font></b>"))
        notificationBuilder.setContentText(Html.fromHtml("<font color='" + getColor(R.color.light) + "'>" + resources.getString(R.string.bindDesc) + "</font>"))
        notificationBuilder.setTicker(resources.getString(R.string.app_name))
        notificationBuilder.setSmallIcon(R.drawable.ic_notification)
        notificationBuilder.setLargeIcon(bitmap)
        notificationBuilder.setAutoCancel(false)
        notificationBuilder.setColor(getColor(R.color.default_color))
        notificationBuilder.setPriority(Notification.PRIORITY_DEFAULT)

        notificationBuilder.setContentIntent(PendingIntent.getActivity(this, 0,
                Intent(this, Configurations::class.java),
                PendingIntent.FLAG_UPDATE_CURRENT))

        return notificationBuilder.build()
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    private fun bindServiceHigh(): Notification {

        val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager?

        val notificationChannel = NotificationChannel(packageName, getString(R.string.app_name), NotificationManager.IMPORTANCE_DEFAULT)

        notificationManager
                ?.createNotificationChannel(notificationChannel)

        val notificationBuilder = Notification.Builder(this)
        notificationBuilder.setContentTitle(Html.fromHtml("<b><font color='" + getColor(R.color.light) + "'>" + resources.getString(R.string.bindTitle) + "</font></b>"))
        notificationBuilder.setContentText(Html.fromHtml("<font color='" + getColor(R.color.light) + "'>" + resources.getString(R.string.bindDesc) + "</font>"))
        notificationBuilder.setTicker(resources.getString(R.string.app_name))
        notificationBuilder.setSmallIcon(R.drawable.ic_notification)
        notificationBuilder.setLargeIcon(BitmapFactory.decodeResource(resources, R.drawable.ic_launcher))
        notificationBuilder.setAutoCancel(false)
        notificationBuilder.setColor(getColor(R.color.default_color))
        notificationBuilder.setChannelId(packageName)

        notificationBuilder.setContentIntent(PendingIntent.getActivity(this, 0,
                Intent(this, Configurations::class.java),
                PendingIntent.FLAG_UPDATE_CURRENT))

        return notificationBuilder.build()
    }
}